ios general/ios general/CoreData/OrganizationMO+Custom.m

```diff
@@ -0,0 +1,54 @@
+#import "OrganizationMO+Custom.h"
+#import "EmployeeMO+Custom.h"
+
+@implementation OrganizationMO (Custom)
+
+- (int)calculateAverageSalary
+{
+    NSNumber *salary = [self.employees valueForKeyPath:@"@avg.salary"];
+    int result = [salary intValue];
+
+    for (EmployeeMO *e in self.employees)
+    {
+        NSLog(@"Employee: %@; Salary: %d", e, e.salary);
+    }
+    NSLog(@"Average salary is: %d", result);
+    return result;
+}
+
+- (EmployeeMO *)employeeWithLowestSalary
+{
+    EmployeeMO *employee = [[EmployeeMO alloc] init];
```

Comment:
It can be rewritten via sorting.
Doesn't seem like better option, I guess?
```objective-c
- (EmployeeMO *)employeeWithLowestSalary
{
    NSManagedObjectContext *context = [AppDelegate shared].managedObjectContext;
    EmployeeMO *employee = [NSEntityDescription insertNewObjectForEntityForName:@"Employee" inManagedObjectContext:context];

    NSSortDescriptor *descriptor = [[NSSortDescriptor alloc] initWithKey:@"salary" ascending:YES];
    NSArray *sortDescriptors = @[descriptor];
    NSArray<EmployeeMO *> *sortedArray = [self.employees sortedArrayUsingDescriptors:sortDescriptors];
    
    employee = sortedArray.firstObject;

    NSLog(@"Employee with the lowest salary: %@, %d", employee, employee.salary);
    return employee;
}
```